exec/native_compile.go

```diff
@@ -91,7 +91,7 @@ func (vm *VM) tryNativeCompile() error {
 	}
 
 	for i := range vm.funcs {
-		if _, isGoFunc := vm.funcs[i].(*goFunction); isGoFunc {
+		if _, isGoFunc := vm.funcs[i].(goFunction); isGoFunc {
```

Comment:
hum...
this kind of type assertion, using the "comma-ok" variant, shouldn't panic.
so I must admit I am not completely clear on what `panic` this should cure.

that said, it seems indeed that we should probably type-assert for `goFunction`:
```
$> git grep goFunction
exec/func.go:38:type goFunction struct {
exec/func.go:43:func (fn goFunction) call(vm *VM, index int64) {
exec/native_compile.go:94:              if _, isGoFunc := vm.funcs[i].(*goFunction); isGoFunc {
exec/native_compile.go:181:             if _, isGoFunc := vm.funcs[i].(*goFunction); isGoFunc {
exec/vm.go:130:                 vm.funcs[i] = goFunction{
```
(as we only ever fill `vm.funcs` with `goFunction{...}` values...)

Reply:
Sorry, my fault for underspecifying - it passes this check, so in the code below it (line 98) it panics trying to convert access it as a compiledFunction

Reply:
Hmmm, I dont understand how we would hit this panic. Consider that `vm.funcs` is of type `[]function`, and `function` is an interface type. The `goFunction` type only implements the necessary interface methods on itself, it does not implement it on its pointer receiver.

As such, I was thinking only `goFunction{}` could implement `function`, not a `*goFunction` ? Now I'm not sure ...

Either way, the current logic is negated, considering we only want to run the loop logic on the func if it is of type `compiledFunction`.

What do you think of a fix like this?

```go
fn, isCompiledFn := vm.funcs[i].(compiledFunction)
if !isCompiledFn {
  continue
}
```